packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -1066,11 +1066,22 @@ const kBuildNumber = 'BuildNumber';
 /// Will be "build" when building and "install" when archiving.
 const kXcodeAction = 'Action';
 
-/// The define of the Xcode build Pre-action.
-///
-/// Will be "PrepareFramework" when copying the Flutter/FlutterMacOS framework
-/// to the BUILT_PRODUCTS_DIR prior to the build.
-const kXcodePreAction = 'PreBuildAction';
+// The define of the Xcode Build Script.
+/// This may be [kPrepareXcodeBuildScript], [kBuildXcodeBuildScript], or [kEmbedXcodeBuildScript].
+const kXcodeBuildScript = 'XcodeBuildScript';
+
+/// When [kXcodeBuildScript] equals this value, that indicates that the target was trigged to run
+/// by a scheme pre-action.
+const kPrepareXcodeBuildScript = 'prepare';
```

Comment:
Could we name these three something like `kXcodeBuildScriptValuePrepare` to make it clearer that there's a key-value relationship between these and the variable above?